### ISSUE TITLE
Blob is not defined

### ISSUE THREAD

**Issue by Olgoetz** [chrisguttandin/web-audio-beat-detector#285]
Hi together,

I am encountering the follow error:

```
var blob = new Blob([worker], {
                   ^

ReferenceError: Blob is not defined
    at /Users/olivergoetz/Development/React-Apps/BPM-Analyzer/node_modules/web-audio-beat-detector/build/es5/bundle.js:10:13
    at worker (/Users/olivergoetz/Development/React-Apps/BPM-Analyzer/node_modules/web-audio-beat-detector/build/es5/bundle.js:2:65)
    at Object.<anonymous> (/Users/olivergoetz/Development/React-Apps/BPM-Analyzer/node_modules/web-audio-beat-detector/build/es5/bundle.js:5:2)
    at Module._compile (module.js:652:30)
    at Object.Module._extensions..js (module.js:663:10)
    at Module.load (module.js:565:32)
    at tryModuleLoad (module.js:505:12)
    at Function.Module._load (module.js:497:3)
    at Module.require (module.js:596:17)
    at require (internal/module.js:11:18)
```

Even when requiring blob, another error with
`URL.createObjectURL is not a function`

Any suggestion to solve this issue I appreciate.

**Comment by chrisguttandin** [comment]
Hi @Olgoetz,

do you maybe run the package with Node.js? web-audio-beat-detector does rely on the Web Audio API which is only available in the browser and therefore it doesn't work with Node.js. A Blob is used internally which is also only available in the browser.

Maybe you are doing some sort of server side rendering. But that is just a wild guess because there is a "React-Apps" folder in your file path. In that case you are maybe able to make sure that web-audio-beat-detector is only executed in the browser.

I hope this helps to resolve the problem.

**Comment by jchiatt** [comment]
I ran into the same issue while using this in a Gatsby site. For anyone else who runs into it in a Gatsby project, you can tell Webpack to ignore this module when Gatsby is building by adding this to your `gatsby-node.js` file:

~~~js
exports.onCreateWebpackConfig = ({ stage, loaders, actions }) => {
  if (stage === "build-html") {
    actions.setWebpackConfig({
      module: {
        rules: [
          {
            test: /bad-module/,
            use: loaders.null(),
          },
        ],
      },
    })
  }
}
~~~

More info: https://www.gatsbyjs.org/docs/debugging-html-builds/

@chrisguttandin I took a quick look at the module to see if I could add check for `window` but I wasn't sure exactly where to put it. In any case, thanks for making this!

**Comment by chrisguttandin** [comment]
I wonder if it would help to instantiate the worker lazily? Currently the `Worker` and the `Blob` are instantiated right away. https://github.com/chrisguttandin/web-audio-beat-detector/blob/master/src/module.ts

Doing this lazily would at least allow to have this module as a dependency in Node.js code. Calling a function would still throw an error though. Would that help?

**Comment by chrisguttandin** [comment]
Thanks to @jgentes, this has been finally fixed by #353.
